### PR TITLE
[TensorIF] Add README.md for tensor_if element @open sesame 9/24 15:25

### PR DESCRIPTION
**Pull request by gichan-jang**
Describe for tensor_if definition, its properties and usage cases.
Related issue: #1703 

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_if/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+# NNStreamer::tensor\_if
+
+## Supported features
+
+GstTensorIF control controls its src-pad based on the values (other/tensor(s)) of its sink-pad.
```
Comment (dongju-chae) [inline gst/nnstreamer/tensor_if/README.md:5]: typo?

Comment (gichan-jang) [reply gst/nnstreamer/tensor_if/README.md]: updated.

**Comment by myungjoo** [comment]
For the documentation. if you have some ideas or remember PL/compiler classes (about parsers), you may suggest more user-friendly properties.

However, for implementation, the first version should implement the most basic (and w/ immediate needs) feature.

**Comment by gichan-jang** [comment]
> For the documentation. if you have some ideas or remember PL/compiler classes (about parsers), you may suggest more user-friendly properties.
> 
> However, for implementation, the first version should implement the most basic (and w/ immediate needs) feature.

Yes. I will start with basic features first. (Passthrough and skip)
I'm considering the structure for feature expansion, should tensor_if element support multiple source pads?
For example, If "then"(true action) picks tensor1, 3 and "else"(false action) picks tensor2, tensor_if should provide multiple source pads. In this case, GstBaseTransform is not support multiple pads. 
So if we need this usecase, I will start writing code without GstBaseTransform.

**Comment by myungjoo** [comment]
> So if we need this usecase, I will start writing code without GstBaseTransform.

Things to consider:
1. can we support them w/ tensor-mux + tensor-if ?
2. how are you going to support dynamic caps? (caps of srcpads will change dynamically according to valus in sinkpad.. would it be manageable? or would it perform highly enough?)

**Review by dongju-chae — APPROVED** [review]
LGTM

**Comment by gichan-jang** [comment]
It is inefficient to calculate the sum or average each time for compared value.
Is it better to set the period to calculate?
e,g,. 1 time per second. or 1 timer per 30 frames.

**Review by myungjoo — APPROVED** [review]

**Pull request merged**